src/dataset.js

```diff
@@ -69,13 +100,21 @@ export default class Dataset {
     }
   }
 
-  setReadOffset(readOffset) {
-    if (this.state.readOffset === readOffset) { return; }
+
+  setReadOffset(readOffset, options = {}) {
```

Comment:
I'll work to organize this better, but there would be SO MUCH duplicate code, that it made more sense to do setReadOffset with options! Over 96% of the code would be duplicate code.